# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1314634451 25200
#      Mon Aug 29 09:14:11 2011 -0700
# Node ID 9a2a6054460f9ca2b53617c18bb64e2177ebcdb1
# Parent  4a86826dbb0e89ac509d34ce46c3e906965fd2aa
maint: Fix 2 compiler warnings

rand.cc: Remove unused variable from function.

load-save.cc: Remove unused initialization in for loop.

diff --git a/src/DLD-FUNCTIONS/rand.cc b/src/DLD-FUNCTIONS/rand.cc
--- a/src/DLD-FUNCTIONS/rand.cc
+++ b/src/DLD-FUNCTIONS/rand.cc
@@ -172,17 +172,16 @@ do_rand (const octave_value_list& args, 
             if (r.all_elements_are_ints ())
               {
                 octave_idx_type n = r.nelem ();
 
                 dims.resize (n);
 
                 octave_idx_type base = NINTbig (r.base ());
                 octave_idx_type incr = NINTbig (r.inc ());
-                octave_idx_type lim = NINTbig (r.limit ());
 
                 for (octave_idx_type i = 0; i < n; i++)
                   {
                     //Negative dimensions are treated as zero for Matlab
                     //compatibility
                     dims(i) = base >= 0 ? base : 0;
                     base += incr;
                   }
diff --git a/src/load-save.cc b/src/load-save.cc
--- a/src/load-save.cc
+++ b/src/load-save.cc
@@ -659,17 +659,18 @@ Force Octave to assume the file is in Oc
   // user preference variable since we can determine the type of file
   // as we are reading.
 
   load_save_format format = LS_UNKNOWN;
 
   bool list_only = false;
   bool verbose = false;
 
-  for (i; i < argc; i++)
+  //for (i; i < argc; i++)
+  for (; i < argc; i++)
     {
       if (argv[i] == "-force" || argv[i] == "-f")
         {
           // Silently ignore this
           // warning ("load: -force ignored");
         }
       else if (argv[i] == "-list" || argv[i] == "-l")
         {
